### PR TITLE
Make the completion service return API response directly

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Skip intermediate transformation to CompletionEvent and return the API
response directly. This approach will be followed by a new implementation
that directly stores to and retrieves from the index instead of using the
transaction stream and filtering it.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala [hunk 1]
```diff
@@ -3,29 +3,68 @@
...
-import com.digitalasset.ledger.api.domain.CompletionEvent.{CommandAccepted, CommandRejected}
+import com.digitalasset.ledger.api.domain.RejectionReason
+import com.digitalasset.ledger.api.domain.RejectionReason.{
+  Disputed,
+  Inconsistent,
+  OutOfQuota,
+  PartyNotKnownOnLedger,
+  SubmitterCannotActViaParticipant,
+  TimedOut
+}
```
Comment (SamirTalwar-DA) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala]: Seems unnecessary to import these twice. (Or at all; I'd prefer prefixing them.)

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala [hunk 2]
```diff
@@ -3,29 +3,68 @@
...
 // TODO - the participant can read completions off the index directly AND
 // TODO - the in-memory sandbox is gone
 private[platform] object CompletionFromTransaction {
 
-  private def toApiOffset(offset: LedgerDao#LedgerOffset): domain.LedgerOffset.Absolute =
-    domain.LedgerOffset.Absolute(Ref.LedgerString.assertFromString(offset.toString))
+  private def toApiCheckpoint(
+      recordedAt: Instant,
+      offset: LedgerDao#LedgerOffset): Option[Checkpoint] =
+    Option(
```
Comment (SamirTalwar-DA) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala]: I would use `Some` here and lower down when it's clear it's a valid object, but not going to complain too much. I'd like to see consistency.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala [hunk 3]
```diff
@@ -3,29 +3,68 @@
...
+      Inconsistent,
+      Disputed,
+      PartyNotKnownOnLedger,
+      OutOfQuota,
+      TimedOut,
+      SubmitterCannotActViaParticipant
+    }
+    rejection match {
+      case _: Inconsistent | _: Disputed | _: PartyNotKnownOnLedger =>
+        Code.INVALID_ARGUMENT.value()
```
Comment (SamirTalwar-DA) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala]: I think you can move the `.value()` call to the call site.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala (hunk 2 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala]: Makes sense, thanks.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala (hunk 3 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala]: Ah, yes, good catch, thanks.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala]: `scalafmt` puts newlines that can reduce the readability but I'll do this, if we don't like it we can always turn back, thanks.

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala]: Addressed by https://github.com/digital-asset/daml/pull/4675/commits/7cedc01cfb3513928288f672123e578ed0d6f4a8

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala (hunk 3 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala]: Addressed by https://github.com/digital-asset/daml/pull/4675/commits/43dccf73f460eed53af535076097b6b51d1e601c

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala (hunk 2 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala]: Addressed by https://github.com/digital-asset/daml/pull/4675/commits/69d2c05532d6197301717fba2a458b909a855af8

**Pull request merged**